launcher/src/main.rs

```diff
@@ -101,17 +110,26 @@ fn resolve_attention(config: &Option<Config>, lora_adapters: &Option<String>) ->
             }
             _ => {
                 if attention.is_none() {
-                    tracing::info!("Forcing flash decoding because head dim is not supported by flashinfer, also disabling prefix caching");
-                    attention = Some("flashdecoding".to_string());
+                    tracing::info!("Forcing attention to '{fallback_attention}' because head dim is not supported by flashinfer, also disabling prefix caching");
+                    attention = Some(fallback_attention.to_string());
                 }
                 if prefix_caching.is_none() {
                     prefix_caching = Some("0".to_string());
                 }
             }
         }
     }
-    let prefix_caching = prefix_caching.unwrap_or("true".to_string());
     let attention = attention.unwrap_or("flashinfer".to_string());
+    let prefix_caching = if attention == "paged"
+        && prefix_caching.is_none()
+        && compute_capability.is_some()
+    {
+        tracing::info!("Disabling prefix caching because it is not supported with 'flashinfer'");
+        "false".to_string()
+    } else {
+        prefix_caching.unwrap_or("true".to_string())
+    };
```

Comment:
Can't we move upstairs ? We're not allowed to change anything if the user sets it directly so this line must always be `unwraps_or`.
And in the previous code every modification must be protected to check that the thing is not None (there might even be a clearer method for it on `Option`.

Reply:
Do we need an `unwrap_or`? The first branch of the conditional only fires when `prefix_caching` is `None`, so the user has set something. Don't know why I added the `compute_capability.is_some()` though.

Reply:
I'll reread the preceding code to see if we check this everywhere.

Reply:
I don't know how to make it the most obvious tbh.

But what I was really aiming for was:

- User signal takes precedence over everything else.
- If not user specifed resolve attention (prefx + attention) as simply as possible:
  - When some conditions are met, we use non defaults.
  - If nothing else happens we use the defaults (so defaults are more easily seen imho)

I wasn't super happy with the design but that's the reason for the big ugly block on top.This is where I hide 'non defauls resolutions' 

So the spirit of my comment is just to try and keep the same rational, have the messy thing only in 1 place. (With the hidden agenda that this mess should disappear, whenever flashinfer supports all the features we would hope it does)